From 84a4230b195f578c43d6e221b4a04f546fd998f9 Mon Sep 17 00:00:00 2001
From: Alexey Tikhonov <atikhono@redhat.com>
Date: Mon, 18 Oct 2021 22:35:13 +0200
Subject: [PATCH 06/17] KCM: fixed uninitialized value
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes following warnings:
```
Error: UNINIT (CWE-457):
sssd-2.6.0/src/responder/kcm/kcmsrv_ccache.c:285: var_decl: Declaring variable "ret" without initializer.
sssd-2.6.0/src/responder/kcm/kcmsrv_ccache.c:323: uninit_use: Using uninitialized value "ret".
 #  321|       krb5_free_context(kctx);
 #  322|
 #  323|->     return ret;
 #  324|   #else
 #  325|       return EOK;

Error: CLANG_WARNING:
sssd-2.6.0/src/responder/kcm/kcmsrv_ccache.c:323:5: warning[core.uninitialized.UndefReturn]: Undefined or garbage value returned to caller
 #  321|       krb5_free_context(kctx);
 #  322|
 #  323|->     return ret;
 #  324|   #else
 #  325|       return EOK;
```

Reviewed-by: Pavel Březina <pbrezina@redhat.com>
---
 src/responder/kcm/kcmsrv_ccache.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/responder/kcm/kcmsrv_ccache.c b/src/responder/kcm/kcmsrv_ccache.c
index ef174e0a090c54b72d488f6d68041d2ac117990d..b63fc70afa35d52e79771d7a5c9f679bdead9f39 100644
--- a/src/responder/kcm/kcmsrv_ccache.c
+++ b/src/responder/kcm/kcmsrv_ccache.c
@@ -294,6 +294,7 @@ kcm_cc_remove_duplicates(struct kcm_ccache *cc,
     kcrd = kcm_cred_to_krb5(kctx, kcm_crd);
     if (kcrd == NULL) {
         DEBUG(SSSDBG_CRIT_FAILURE, "Failed to convert kcm cred to krb5\n");
+        ret = ERR_INTERNAL;
         goto done;
     }
 
@@ -301,6 +302,7 @@ kcm_cc_remove_duplicates(struct kcm_ccache *cc,
         kcrd_cc = kcm_cred_to_krb5(kctx, p);
         if (kcrd_cc == NULL) {
             DEBUG(SSSDBG_CRIT_FAILURE, "Failed to convert kcm cred to krb5\n");
+            ret = ERR_INTERNAL;
             goto done;
         }
 
-- 
2.31.1

